vkrunner/vr-pipeline.c

```diff
@@ -770,10 +865,17 @@ vr_pipeline_create(const struct vr_config *config,
         pipeline->stages = get_script_stages(script);
 
         if (script->n_buffers > 0) {
-                pipeline->descriptor_set_layout =
-                        create_vk_descriptor_set_layout(pipeline, script);
-                if (pipeline->descriptor_set_layout == VK_NULL_HANDLE)
+                if (!create_vk_descriptor_set_layout(pipeline, script) ||
+                    !pipeline->descriptor_set_layout ||
+                    !pipeline->n_desc_sets) {
                         goto error;
+                } else {
+                        for (unsigned i = 0; i < pipeline->n_desc_sets; i++) {
+                                if (pipeline->descriptor_set_layout[i] ==
+                                    VK_NULL_HANDLE)
+                                        goto error;
```

Comment:
I’m still not sure why there are all these extra error checks to verify that `descriptor_set_layout` and all of its entries are not `NULL`. If any of those were the case then `create_vk_descriptor_set_layout` would have returned `false`, right? It seems better to avoid duplicating the error checking and encapsulate it in the function that handles it.